common/Windows/FileDialog/WindowOpenFolderDialog.cs

```diff
@@ -28,11 +31,21 @@ protected override void InitializeInstance()
 
     public async Task<StorageFolder?> ShowAsync(Window window)
     {
-        if (ShowOk(window))
+        try
+        {
+            if (ShowOk(window))
+            {
+                FileDialog.GetResult(out var shellItem);
+                var folderPath = GetDisplayName(shellItem);
+
+                // GetFolderFromPathAsync will throw if the user does not have access to the
+                // folder.  One such example is a hidden folder.
+                return await StorageFolder.GetFolderFromPathAsync(folderPath).AsTask();
+            }
+        }
+        catch (Exception ex)
         {
-            FileDialog.GetResult(out var shellItem);
-            var folderPath = GetDisplayName(shellItem);
-            return await StorageFolder.GetFolderFromPathAsync(folderPath).AsTask();
+            _log.Error(ex.ToString());

```

Comment:
Do we want to treat all of these as Errors? If a user doesn't have access, throwing that exception is Dev Home behaving correctly, so I'm not sure we want to log as Error (warning maybe?)

Reply:
> Do we want to treat all of these as Errors? If a user doesn't have access, throwing that exception is Dev Home behaving correctly, so I'm not sure we want to log as Error (warning maybe?)

Also, if a user doesn't have access, how do we communicate the nature of the failure to the user? Is there something doing that here?

Reply:
> how do we communicate the nature of the failure to the user? 
I believe this would need design input on where to put the error message.  

Currently, the user does not have a way to understand why the operation did not work.

Reply:
>  throwing that exception is Dev Home behaving correctly,
The purpose of this PR is to prevent DevHome crashing in the event a user points to a hidden directory from the File Explorer integration menu.

Instead of a try/catch here, I can add it in the code that does not handle the exception.  That will prevent the crash.

Reply:
This is moved to FileExplorerViewModel.cs.  This localizes the try/catch.  Should be enough for this PR to avoid a crash.